lib/components/map/connected-transitive-overlay.js

```diff
@@ -1,71 +1,43 @@
-import isEqual from 'lodash.isequal'
-import coreUtils from '@opentripplanner/core-utils'
-import TransitiveCanvasOverlay from '@opentripplanner/transitive-overlay'
+import TransitiveOverlay from '@opentripplanner/transitive-overlay'
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
-import { ComponentContext } from '../../util/contexts'
-import { getActiveItinerary, getActiveItineraries, getActiveSearch } from '../../util/state'
+import { getActiveSearch, getTransitiveData } from '../../util/state'
 
 /**
- * Wrapper for TransitiveCanvasOverlay that passes getTransitiveRouteLabel defined in ComponentContext.
+ * Wrapper for TransitiveOverlay that avoids rerenders by checking transitive
+ * data computed from redux store.
  */
-class TransitiveCanvasOverlayWithContext extends Component {
-  static contextType = ComponentContext
-
+class TransitiveCanvasOverlay extends Component {
   shouldComponentUpdate (nextProps) {
-    return (
-      !isEqual(nextProps.activeSearchVisibleItinerary, this.props.activeSearchVisibleItinerary) ||
-      !isEqual(nextProps.activeSearchResponseOtp, this.props.activeSearchResponseOtp)
-    )
+    return nextProps.transitiveData !== this.props.transitiveData
```

Comment:
The object references can be different although the value and contents can be identical. I thought in such cases a rendering could be avoided.